lib/decanter/collection_detection.rb

```diff
@@ -0,0 +1,24 @@
+module Decanter
+  module CollectionDetection
+    def self.included(base)
+      base.extend(ClassMethods)
+    end
+
+    module ClassMethods
+      def decant(args, **options)
+        return super(args) unless collection?(args, options[:is_collection])
+
+        args.map { |resource| super(resource) }
+      end
+
+      private
+
+      # leveraging the approach used in the [fast JSON API gem](https://github.com/Netflix/fast_jsonapi#collection-serialization)
+      def collection?(args, collection_option = nil)
+        return collection_option == true unless collection_option.nil?
```

Comment:
Boolean methods should return true or false, so I'm casting this here instead of just returning the `collection_option`. Alternatively I could coerce it using `!!` or apply a different coercion strategy.

Reply:
yeah this is tricky, and I am shaking my head at the fact that I wasn't able to catch how I wrote this incorrectly before.

That being said, there's a couple decision points (related to your readme above.). If we want to allow truthy values as you indicate above then you would want to use `!!` (then `yes` would follow the `true` option rather than the `false` option). That does feel unexpected though, and I'd prefer this option only takes a true boolean. Thoughts on raising for a non-accepted value?